src/ert/runpaths.py

```diff
@@ -88,12 +90,19 @@ def write_runpath_list(
         with open(self.runpath_list_filename, "w", encoding="utf-8") as filehandle:
             for iteration in iteration_numbers:
                 for realization in realization_numbers:
-                    job_name = self._substitution_list.substitute_real_iter(
+                    job_name_or_eclbase = self._substitution_list.substitute_real_iter(
                         self._jobname_format, realization, iteration
                     )
                     runpath = self._substitution_list.substitute_real_iter(
                         self._runpath_format, realization, iteration
                     )
+                    if self._eclbase is not None:
+                        job_name_or_eclbase = (
+                            self._substitution_list.substitute_real_iter(
+                                self._eclbase, realization, iteration
+                            )
+                        )
+
```

Comment:
```
eclbase_or_job_name = self._eclbase if self._eclbase else self._jobname_format

job_name_or_eclbase = self._substitution_list.substitute_real_iter(
                       eclbase_or_job_name, realization, iteration
)
```

Alternatively something like this? Less code-dedup?

Reply:
Did a similar thing